internal/emigration/emigrate-nobles.lua

```diff
@@ -0,0 +1,352 @@
+--@module = true
+
+--Deport resident nobles of other lands and summon your rightful lord if he/she is elsewhere
+
+--[[
+TODO:
+  * Feature: have rightful ruler immigrate to fort if off-site
+  * QoL: make sure items are unassigned
+]]--
+
+local argparse = require("argparse")
+
+local options = {
+    all = false,
+    unitId = -1,
+    list = false
+}
+
+local fort = nil      ---@type df.world_site
+local playerCiv = nil ---@type df.historical_entity
+local capital = nil   ---@type df.world_site
+
+-- adapted from Units::get_land_title()
+---@return df.world_site|nil
+local function findSiteOfRule(np)
+    local site = nil
+    local civ = np.entity -- lawmakers seem to be all civ-level positions
+    for _, link in ipairs(civ.site_links) do
+        if not link.flags.land_for_holding then goto continue end
+        if link.position_profile_id ~= np.assignment.id then goto continue end
+
+        site = df.world_site.find(link.target)
+        break
+        ::continue::
+    end
+
+    return site
+end
+
+---@return df.world_site|nil
+local function findCapital(civ)
+    local civCapital = nil
+    for _, link in ipairs(civ.site_links) do
+        if link.flags.capital then
+            civCapital = df.world_site.find(link.target)
+            break
+        end
+    end
+
+    return civCapital
+end
+
+local function addNobleOfOtherSite(unit, nobleList)
```

Comment:
much of this code is copied from `emigration.lua`. could you refactor so the logic is not duplicated?